pkg/csi/service/wcpguest/controller.go

```diff
@@ -452,6 +467,91 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 	timeoutSeconds := int64(getAttacherTimeoutInMin(ctx) * 60)
 	timeout := time.Now().Add(time.Duration(timeoutSeconds) * time.Second)
 
+	var isFileVolume bool
+	// Retrieve Supervisor PVC
+	svPVC, err := c.supervisorClient.CoreV1().PersistentVolumeClaims(c.supervisorNamespace).Get(ctx, req.VolumeId, metav1.GetOptions{})
+	if err != nil {
+		msg := fmt.Sprintf("failed to retrieve supervisor PVC %q in %q namespace. Error: %+v", req.VolumeId, c.supervisorNamespace, err)
+		log.Error(msg)
+		return nil, status.Error(codes.Internal, msg)
+	}
+	for _, accessMode := range svPVC.Spec.AccessModes {
+		if accessMode == v1.ReadWriteMany || accessMode == v1.ReadOnlyMany {
+			isFileVolume = true
+		}
+	}
+	// Check the feature state for file volume support
+	if isFileVolume {
+		if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FileVolume) {
+			// Retrieve Supervisor PVC
+
+			cfg, err := common.GetConfig(ctx)
+			if err != nil {
+				msg := fmt.Sprintf("failed to read config. Error: %+v", err)
+				log.Error(msg)
+				return nil, status.Errorf(codes.Internal, msg)
+			}
+			restClientConfig := k8s.GetRestClientConfigForSupervisor(ctx, cfg.GC.Endpoint, cfg.GC.Port)
```

Comment:
Can we avoid creating the client for every `ControllerUnpublishVolume` callback?

Reply:
Done